lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -265,6 +256,52 @@ defmodule BroadwayRabbitMQ.AmqpClient do
       {:error, :timeout}
   end
 
+  defp get_channel(%{connection: {:custom_pool, module, args}}) do
+    with {:ok, channel} <- module.checkout_channel(args),
+         true <- Process.link(channel.pid) do
+      {:ok, channel}
+    end
```

Comment:
Let's make sure that the return value of the `checkout_channel/1` callback is correct, otherwise we risk bubbling unexpected stuff up, which can turn out hard to debug.

```suggestion
    case module.checkout_channel(args) do
      {:ok, %AMQP.Channel{} = channel} ->
        true = Process.link(channel.pid)
        {:ok, channel}
        
      {:error, exception} when is_exception(exception) ->
        {:error, exception}
        
      other ->
        raise """
        expected #{Exception.format_mfa(module, :checkout_channel, 1)} to \
        return {:ok, AMQP.Channel.t()} or {:error, exception}, got: \
        #{inspect(other)}\
        """
    end
```